tests/components/gdacs/test_geo_location.py

```diff
@@ -0,0 +1,242 @@
+            ATTR_EXTERNAL_ID: "2345",
+            ATTR_LATITUDE: 38.1,
+            ATTR_LONGITUDE: -3.1,
+            ATTR_FRIENDLY_NAME: "Tropical Cyclone: Name 2",
+            ATTR_DESCRIPTION: "Description 2",
+            ATTR_EVENT_TYPE: "Tropical Cyclone",
+            ATTR_UNIT_OF_MEASUREMENT: "km",
+            ATTR_SOURCE: "gdacs",
+            ATTR_ICON: "mdi:weather-hurricane",
+        }
+        assert float(state.state) == 20.5
+
+        state = hass.states.get("geo_location.tropical_cyclone_name_3")
+        assert state is not None
+        assert state.name == "Tropical Cyclone: Name 3"
+        assert state.attributes == {
+            ATTR_EXTERNAL_ID: "3456",
+            ATTR_LATITUDE: 38.2,
+            ATTR_LONGITUDE: -3.2,
+            ATTR_FRIENDLY_NAME: "Tropical Cyclone: Name 3",
+            ATTR_DESCRIPTION: "Description 3",
+            ATTR_EVENT_TYPE: "Tropical Cyclone",
+            ATTR_COUNTRY: "Country 2",
+            ATTR_UNIT_OF_MEASUREMENT: "km",
+            ATTR_SOURCE: "gdacs",
+            ATTR_ICON: "mdi:weather-hurricane",
+        }
+        assert float(state.state) == 25.5
+
+        # Simulate an update - two existing, one new entry, one outdated entry
+        mock_feed_update.return_value = "OK", [mock_entry_1, mock_entry_4, mock_entry_3]
+        async_fire_time_changed(hass, utcnow + DEFAULT_SCAN_INTERVAL)
+        await hass.async_block_till_done()
+
+        all_states = hass.states.async_all()
+        assert len(all_states) == 4
+
+        # Simulate an update - empty data, but successful update,
+        # so no changes to entities.
+        mock_feed_update.return_value = "OK_NO_DATA", None
+        async_fire_time_changed(hass, utcnow + 2 * DEFAULT_SCAN_INTERVAL)
+        await hass.async_block_till_done()
+
+        all_states = hass.states.async_all()
+        assert len(all_states) == 4
+
+        # Simulate an update - empty data, removes all entities
+        mock_feed_update.return_value = "ERROR", None
+        async_fire_time_changed(hass, utcnow + 3 * DEFAULT_SCAN_INTERVAL)
+        await hass.async_block_till_done()
+
+        all_states = hass.states.async_all()
+        assert len(all_states) == 1
+
+
+async def test_setup_imperial(hass):
+    """Test the setup of the integration using imperial unit system."""
+    hass.config.units = IMPERIAL_SYSTEM
+    # Set up some mock feed entries for this test.
+    mock_entry_1 = _generate_mock_feed_entry(
+        "1234",
+        "Description 1",
+        15.5,
+        (38.0, -3.0),
+        event_name="Name 1",
+        event_type_short="DR",
+        event_type="Drought",
+    )
+
+    # Patching 'utcnow' to gain more control over the timed update.
+    utcnow = dt_util.utcnow()
+    with patch("homeassistant.util.dt.utcnow", return_value=utcnow), patch(
+        "aio_georss_client.feed.GeoRssFeed.update", new_callable=CoroutineMock
+    ) as mock_feed_update, patch(
+        "aio_georss_client.feed.GeoRssFeed.__init__",
```

Comment:
Normally we don't patch the init method but the class.

Reply:
Yes, I remember that we had a similar discussion in one my previous PRs. I will look into finding a better solution for this test case tonight.

Reply:
The only alternative I could find is to access some protected members.